### PR TITLE
xenvm: switch to boot header 3

### PR DESCRIPTION
**Pull request by arminn**
Switch to boot header 3 :
* Use vendor_boot
* Intended to use with Android u-boot

### CODE REVIEW

**Comment by arminn** [comment]
@svlad-90 plz review

File: BoardConfig.mk [hunk 1]
```diff
@@ -100,11 +104,21 @@ BOARD_VENDOR_SEPOLICY_DIRS += device/xen/xenvm/sepolicy/vendor
 BOARD_PLAT_PRIVATE_SEPOLICY_DIR += device/xen/xenvm/sepolicy/private
 BOARD_PLAT_PUBLIC_SEPOLICY_DIR += device/xen/xenvm/sepolicy/public
 
+# Boot image build rules
+BOARD_BOOT_HEADER_VERSION := 3
+BOARD_MKBOOTIMG_ARGS += --header_version $(BOARD_BOOT_HEADER_VERSION)
+
+# Turn off AVB so that trout can boot
```
Comment (svlad-90) [inline BoardConfig.mk]: Hi @arminn, you are not using trout, right? If so, please, change the comment.

File: build/device.mk [hunk 2]
```diff
@@ -129,6 +129,9 @@ ifneq ($(DISABLE_AVB),true)
     device/xen/xenvm/fstab.xenvm.avb:root/fstab.xenvm \
     device/xen/xenvm/fstab.xenvm.avb:$(TARGET_COPY_OUT_VENDOR)/etc/fstab.xenvm \
     device/xen/xenvm/fstab.xenvm.avb:$(TARGET_COPY_OUT_RECOVERY)/root/first_stage_ramdisk/fstab.xenvm \
+    device/xen/xenvm/fstab.xenvm.avb:$(TARGET_COPY_OUT_RAMDISK)/first_stage_ramdisk/fstab.xenvm \
```
Comment (svlad-90) [inline build/device.mk]: Hi, @arminn, do you really need the same fstab in all those 3 places?

File: BoardConfig.mk [hunk 3]
```diff
@@ -57,12 +57,16 @@ BOARD_FLASH_BLOCK_SIZE := 512
...
 BOARD_USERDATAIMAGE_PARTITION_SIZE := 3145728000  # 3145728000 = 3000 MiB
 
+BOARD_INCLUDE_DTB_IN_BOOTIMG := false
+PRODUCT_COPY_FILES += device/google/cuttlefish/dtb.img:dtb.img \
+
 BOARD_BUILD_SYSTEM_ROOT_IMAGE := false
-BOARD_USES_RECOVERY_AS_BOOT := true
 BOARD_BUILD_SUPER_IMAGE_BY_DEFAULT := true
 
+
```
Comment (svlad-90) [inline BoardConfig.mk]: Hi @arminn, redundant change.

File: BoardConfig.mk [hunk 4]
```diff
@@ -57,12 +57,16 @@ BOARD_FLASH_BLOCK_SIZE := 512
 BOARD_VENDORIMAGE_FILE_SYSTEM_TYPE := ext4
 BOARD_SYSTEMIMAGE_FILE_SYSTEM_TYPE := ext4
 BOARD_BOOTIMAGE_PARTITION_SIZE := 30720000
+BOARD_VENDOR_BOOTIMAGE_PARTITION_SIZE := 30720000
 BOARD_USERDATAIMAGE_PARTITION_SIZE := 3145728000  # 3145728000 = 3000 MiB
 
+BOARD_INCLUDE_DTB_IN_BOOTIMG := false
+PRODUCT_COPY_FILES += device/google/cuttlefish/dtb.img:dtb.img \
```
Comment (svlad-90) [inline BoardConfig.mk:67]: Hi @arminn, you've explained to me the motivation for the addition of this line. Could you, please, add the same explanation to the commit message? The presence of this line is not self-explaining, thus, requires additional "decoding". ))

Comment (arminn) [reply BoardConfig.mk:67]: Done

File: BoardConfig.mk (hunk 1 above)
Comment (arminn) [reply BoardConfig.mk]: Done

File: build/device.mk (hunk 2 above)
Comment (arminn) [reply build/device.mk]: Removed not needed parts

File: BoardConfig.mk (hunk 3 above)
Comment (arminn) [reply BoardConfig.mk]: Done

**Comment by svlad-90** [comment]
Hi @arminn,

```Reviewed-by: Goncharuk Vladyslav <goncharuk_vladyslav@epam.com>```

**Comment by arminn** [comment]
Added RB

**Pull request merged**